data_pump/bitstream.py

```diff
@@ -0,0 +1,153 @@
+import logging
+import json
+
+from utils import read_json, convert_response_to_json, do_api_post
+
+
+def import_bitstream(metadata_class, bitstreamformat_id_dict, primary_bitstream_dict,
+                     bitstream2bundle_dict, bundle_id_dict,
+                     community2logo_dict, collection2logo_dict, bitstream_id_dict,
+                     community_id_dict, collection_id_dict, unknown_format_id_val,
+                     statistics_dict):
+    """
+    Import data into database.
+    Mapped tables: bitstream, bundle2bitstream, metadata, most_recent_checksum
+    and checksum_result
+    """
+    bitstream_json_name = 'bitstream.json'
+    bundle2bitstream_json_name = 'bundle2bitstream.json'
+    bitstream_url = 'clarin/import/core/bitstream'
+    checksum_url = 'clarin/import/core/bitstream/checksum'
+    imported = 0
+
+    # load bundle2bitstream
+    bundle2bitstream_json_a = read_json(bundle2bitstream_json_name)
+    if bundle2bitstream_json_a:
+        for i in bundle2bitstream_json_a:
+            bitstream2bundle_dict[i['bitstream_id']] = i['bundle_id']
+
+    # load and import bitstreams
+    bitstream_json_a = read_json(bitstream_json_name)
+    if not bitstream_json_a:
+        logging.info("Bitstream JSON is empty.")
+        return
+    counter = 0
+    for i in bitstream_json_a:
+        if counter % 500 == 0:
+            # do bitstream checksum
+            # do this after every 500 imported bitstreams,
+            # because the server may be out of memory
+            # fill the tables: most_recent_checksum and checksum_result
+            # based on imported bitstreams
+            try:
+                do_api_post(checksum_url, None, None)
```

Comment:
second argument should be {} not None. Method expects dictionary, IDE complains.